### PR TITLE
Move defaults and types to constants

### PR DESCRIPTION
**Pull request by pan-x-c**
---
name: Move defaults and types into constants
---
## Description
1. Move default values and  types into a single file and avoid duplication of definitions
2. rename _Operator to Operator
3. fix the typing issue in pipeline package
4. optimize  the order of parameters in functional pipeline
5. update docstring and sphinx doc

## Checklist

Please check the following items before code is ready to be reviewed.

- [x]  Code has passed all tests
- [x]  Docstrings have been added/updated in Google Style
- [x]  Documentation has been updated
- [x]  Code is ready for review

### CODE REVIEW

File: src/agentscope/constants.py [hunk 1]
```diff
@@ -1,4 +1,68 @@
...
+
+
+class ServiceExecStatus(IntEnum):
+    """Enum for service execution status."""
+
+    SUCCESS = 1
+    ERROR = -1
+
+
+class PromptType(IntEnum):
```
Comment (DavdGao) [inline src/agentscope/constants.py:57]: We should take the usage into consideration, e.g. when using PromptType, `agentscope.prompt.PromptType` is better than `agentscope.constants.PromptType`?

File: src/agentscope/pipelines/functional.py [hunk 2]
```diff
@@ -41,24 +41,24 @@ def sequentialpipeline(
 
 
 def ifelsepipeline(
-    x: dict,
     condition_func: Callable,
-    if_body_operator: _Operator,
-    else_body_operator: _Operator = placeholder,
+    if_body_operator: Operator,
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:45]: Similar with `sequentialpipeline`, support `Sequence[Operator]`

File: src/agentscope/pipelines/functional.py [hunk 3]
```diff
@@ -70,27 +70,27 @@ def ifelsepipeline(
 
 
 def switchpipeline(
-    x: dict,
-    condition_func: Callable[[dict], Any],
-    case_operators: Mapping[Any, _Operator],
-    default_operator: _Operator = placeholder,
+    condition_func: Callable[[Any], Any],
+    case_operators: Mapping[Any, Operator],
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:74]: The same as above

File: src/agentscope/pipelines/functional.py [hunk 4]
```diff
@@ -103,28 +103,30 @@ def switchpipeline(
 
 
 def forlooppipeline(
-    x: dict,
-    loop_body_operator: _Operator,
+    loop_body_operator: Operator,
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:106]: The same as above

File: src/agentscope/pipelines/functional.py [hunk 5]
```diff
@@ -135,24 +137,27 @@ def forlooppipeline(
 
 
 def whilelooppipeline(
-    x: dict,
-    loop_body_operator: _Operator,
-    condition_func: Callable[[int, dict], bool] = lambda _, __: False,
+    loop_body_operator: Operator,
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:140]: The same as above

File: src/agentscope/pipelines/functional.py [hunk 6]
```diff
@@ -103,28 +103,30 @@ def switchpipeline(
...
             based on the output of the loop_body_operator, defaults to
             `lambda _: False`
+        x (`Optional[dict]`, defaults to `None`):
+            The input dictionary.
 
     Returns:
         `dict`: The output dictionary.
     """
+    if x is None:
+        x = {}
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:129]: Cannot set `x` to `{}`, for it'll be still added into agent memory.

File: src/agentscope/pipelines/functional.py [hunk 7]
```diff
@@ -135,24 +137,27 @@ def forlooppipeline(
...
             on the current loop number and output of the loop_body_operator,
             defaults to `lambda _,__: False`
+        x (`Optional[dict]`, defaults to `None`):
+            The input dictionary.
 
     Returns:
         `dict`: the output dictionary.
     """
+    if x is None:
+        x = {}
```
Comment (DavdGao) [inline src/agentscope/pipelines/functional.py:160]: Same as above

File: src/agentscope/constants.py (hunk 1 above)
Comment (DavdGao) [reply src/agentscope/constants.py:57]: The same problem with `ServiceExecStatus`

**Review by DavdGao — COMMENTED** [review]
please see inline comments

File: src/agentscope/pipelines/functional.py [hunk 8]
```diff
@@ -40,123 +41,133 @@ def sequentialpipeline(
...
+    else:
+        return operators(x)
+
+
 def ifelsepipeline(
-    x: dict,
     condition_func: Callable,
-    if_body_operator: _Operator,
-    else_body_operator: _Operator = placeholder,
+    if_body_operators: Operators,
```
Comment (qbc2016) [inline src/agentscope/pipelines/functional.py:55]: Do we need to add Sequence[Operators] to the type here?

Comment (pan-x-c) [reply src/agentscope/pipelines/functional.py:55]: `Operators` is an alias for `Union[Operator, Sequence[Operator]]`

**Review by qbc2016 — APPROVED** [review]
LGTM.

**Pull request merged**